documentatie/flowcharts/date-entry-page.md

```diff
@@ -0,0 +1,167 @@
+1. index of menu-item == furthest: `unsubmitted` (blue dot)
+1. page contains errors: error: `error` icon
+1. page contains warnings and warnings have been accepted: `warning` (warning icon)
+1. submitted data was empty `empty` (dash icon)
+1. submitted data was not empty `valid` (green check icon)
+
+### Diagram
+
+```mermaid
+flowchart TD
+    flow-start([start])
+
+    flow-start --> page-visited{page-visited?}
+    page-visited -- no --> disabled-styling
+    page-visited -- yes --> current-page{current-page?}
+    current-page{current-page?} -- yes --> current-styling
+    current-page -- no --> submitted{submitted?}
+    submitted -- no --> unsubmitted-styling
+    submitted -- yes --> errors{errors?}
+    errors -- yes --> error-styling
+    errors -- no --> warnings{warnings?}
+    warnings -- yes --> warning-styling
+    warnings -- no --> empty{empty? }
+    empty -- yes --> empty-styling
+    empty -- no --> green-check-styling
+
+```
+
+
+## Render form
+
+Render happens based on last received API response.
+
+```mermaid
+flowchart TD
+
+    %% elements
+    flow-start([start])
+    flow-end([end])
+    illegal-state{{illegal state}}
+
+    show-error
+    show-warning
+
+    error-any-prev-page{error-any-prev-page?}
+    error-cur-page{error-cur-page?}
+
+    warning-cur-page{warning-cur-page?}
+
+    cur-page-submitted-error{cur-page-submitted?}
+    cur-page-submitted-warning{cur-page-submitted?}
+
+    fill-cached-input
+
+    %% flow
+    flow-start --> error-any-prev-page
+
+    error-any-prev-page -- yes --> illegal-state
+    error-any-prev-page -- no -->  error-cur-page
+
+    error-cur-page -- yes --> cur-page-submitted-error
+    cur-page-submitted-error -- yes --> show-error
+    cur-page-submitted-error -- no --> warning-cur-page
+    error-cur-page -- no --> warning-cur-page
+
+    warning-cur-page -- no --> fill-cached-input
+    warning-cur-page -- yes --> cur-page-submitted-warning
+    
+    cur-page-submitted-warning -- yes --> show-warning
+    cur-page-submitted-warning -- no --> fill-cached-input
+
+    show-error --> fill-cached-input
+    show-warning --> fill-cached-input
+
+    fill-cached-input --> flow-end
+
+```
+
+
+## Click "Volgende"
+
+```mermaid
+flowchart TD
+
+    %% elements
+    flow-start([start])
+    flow-end([end])
+
+    click-next
+    call-api
+    go-to-prev-page
+    go-to-next-page
+    render-cur-page
+
+    error-any-prev-page{error-any-prev-page?}
+    error-cur-page{error-cur-page?}
+    warning-cur-page{warning-cur-page?}
+
+    %% flow
+    flow-start --> click-next
+    click-next --> call-api
+    call-api --> error-any-prev-page
+    error-any-prev-page -- yes --> go-to-prev-page
+    go-to-prev-page --> flow-end
+
+    error-any-prev-page -- no --> error-cur-page
+
+    error-cur-page -- yes --> render-cur-page
+    error-cur-page -- no --> warning-cur-page
+
+    warning-cur-page -- yes --> render-cur-page
```

Comment:
this should be extended with the 'accept warning' flow. 

when current page contains a warning, current page contains a checkbox to accept the warning
if checkbox is checked, the warning is dismissed and user navigates to next page

Reply:
done